data/wpa_supplicant/wpa_supplicant_test.sh

```diff
@@ -93,10 +98,12 @@ ip addr del 192.168.202.2/24 dev wlan1
 echo "Checking wicked/dhcp on wifi interface ... "
 echo 'start dnsmasq' >>hostapd.com
 cp ifcfg-wlan1-dhcp /etc/sysconfig/network/ifcfg-wlan1
-echo "reloading wicked ... "
-systemctl reload wicked
+echo "Restarting wicked ... "
+systemctl restart wicked
+sleep 20
```

Comment:
You are using `sleep()` or are simulating a sleep-like functionality. This is introducing the risk of either unnecessarily wasting time or just shifting race conditions and making them even harder to investigate. You *can* use sleep but there are only very limited cases where is the best solution (e.g. yield threads in low-level system code). Here you have the classical problem in software design of [Synchronization](https://en.wikipedia.org/wiki/Synchronization_(computer_science)) . sleep should not be used in place of proper synchronization methods. https://blogs.msmvps.com/peterritchie/2007/04/26/thread-sleep-is-a-sign-of-a-poorly-designed-program/ is one of multiple references that explains this: "The original problem is likely a timing/synchronization issue, ignoring it by hiding it with Thread.Sleep is only going to delay the problem and make it occur in random, hard to reproduce ways." And if the problem does not occur at least time is wasted.

We are waiting because some condition changes some time ... keyword(s) is/are *some time*. It is important to know the exact condition that the code should wait for and check for that instead of delaying program execution for an arbitrary amount of time. For openQA tests in general just try to do what you as a human user would also do. You would not look on your watch waiting for exactly 100s before you look at the screen again, right? :)

How about `until journalctl -e -u wicked | grep -q 'wicked restarted'; do sleep .1; done` or whatever expression you actually wait for here?

Reply:
I am fully aware of this issue but this PR is a very much needed hotfix, as Clemens' work is blocked by it atm.
Knowing that this is bad design and should be fixed ASAP I would still recommend to proceed and fix this once Clemens' work on `wicked` is done.

Is that ok with you @okurz @cfconrad ?

Reply:
I'm ok with sleep, as we will kick it anyway. The wicked-wlan tests will be implemented in the wicked-testsuite, see https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/11440
I'm currently on enabling these tests.